### PR TITLE
(COMPL): Autocomplete break and continue keywords

### PR DESCRIPTION
**Pull request by alygin**
Autocompletion for the `break` and `continue` keywords inside loops.

### CODE REVIEW

**Comment by alygin** [comment]
Tests a broken because of the backtraces setting rollback.

File: src/main/kotlin/org/rust/lang/core/RustPsiPattern.kt [hunk 1]
```diff
@@ -80,6 +81,12 @@ object RustPsiPattern {
     val onTestFn: PsiElementPattern.Capture<PsiElement> = onItem(psiElement<RustFunctionElement>()
         .withChild(psiElement<RustOuterAttrElement>().withText("#[test]")))
 
+    val inAnyLoop: PsiElementPattern.Capture<PsiElement> =
+        psiElement().inside(true, psiElement<RustBlockElement>().withParent(or(
```
Comment (matklad) [inline src/main/kotlin/org/rust/lang/core/RustPsiPattern.kt]: One possible improvement in performance and correctness is to try to stop earlier when searching for parent. That is, `break` should not be suggested inside a lambda inside a loop.

**Comment by matklad** [comment]
LGTM, I've restarted the travis build

File: src/main/kotlin/org/rust/lang/core/RustPsiPattern.kt (hunk 1 above)
Comment (alygin) [inline src/main/kotlin/org/rust/lang/core/RustPsiPattern.kt]: Sure, you're right. I'll add a proper `stopAt`.

**Comment by matklad** [comment]
Thanks

**Pull request merged**